### ISSUE TITLE
`dotnet publish` creating wrong executable name

### ISSUE THREAD

**Issue by sandyarmstrong** [dotnet/sdk#1396]
Tested on OS X 10.12.5 with .NET Core 2.0.0-preview2-006497.

## Steps to reproduce

1. `mkdir PublishTest.WithDot && cd PublishTest.WithDot` (must be a dotted name to exhibit bug)
2. `dotnet new console`
3. Add `<RuntimeIdentifiers>win-x64;osx.10.11-x64</RuntimeIdentifiers>` in the default `PropertyGroup` in the generated csproj.
4. `dotnet build && dotnet publish -r osx.10.11-x64`
5. `ls bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot*`

## Expected Results

```
bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot
bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot.deps.json
bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot.dll
bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot.pdb
bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot.runtimeconfig.json
```

## Actual Results

```
bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot.WithDot
bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot.deps.json
bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot.dll
bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot.pdb
bin/Debug/netcoreapp2.0/osx.10.11-x64/publish/PublishTest.WithDot.runtimeconfig.json
```

The incorrectly-named executable does not work until it is renamed:

```bash
$ cd bin/Debug/netcoreapp2.0/osx.10.11-x64/publish

$ chmod +x PublishTest.WithDot.WithDot

$ ./PublishTest.WithDot.WithDot
The managed DLL bound to this executable: 'PublishTest.WithDot.dll', did not match own name 'PublishTest.WithDot.WithDot.dll'.
A fatal error was encountered. This executable was not bound to load a managed DLL.

$ mv PublishTest.WithDot.WithDot PublishTest.WithDot

$ ./PublishTest.WithDot
Hello World!
```

**Comment by sandyarmstrong** [comment]
This is a regression since the 1.0.4 SDK.

**Comment by rainersigwald** [comment]
I believe this is a duplicate of dotnet/cli#6397, fixed in the latest builds by #1378.
